Fix BL-3332 so Leveled reader external links work. Remove need for JS

hatton opened the pull request:
Now there is a proper api controller for exeternal links.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloomdesktop/1051)
<!-- Reviewable:end -->

StephenMcConnel commented:
Generally looks okay, but a couple of questions/comments to address.  :-)

---

Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/BloomBrowserUI/bookEdit/toolbox/readers/leveledReader/leveledReaderToolboxPanel.jade, line 64 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-KH5MesIRO_HR7u_ZQz0:-KH5MesIRO_HR7u_ZQz1:-1731724243) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e049888b50a473aacddc927b642ddf6403442e41/src/BloomBrowserUI/bookEdit/toolbox/readers/leveledReader/leveledReaderToolboxPanel.jade#L64)):*
Is the &test=foo&blah=boo actually needed here?  what purpose do they serve?

---

*[src/BloomExe/web/controllers/ExternalLinkController.cs, line 27 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-KH5UusQEWWLAJk4ymxi:-KH5UusRqaHKwUdx5sXN:391064277) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e049888b50a473aacddc927b642ddf6403442e41/src/BloomExe/web/controllers/ExternalLinkController.cs#L27)):*
It's probably okay as is, but I would prefer HandleRequest as the name of the method.  Handle initially brings up the image of a magic binary number in my mind.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-:-KH5XEU6lVW6Yskpp8t1:-160408394)*
<!-- Sent from Reviewable.io -->

JohnThomson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, with minor comments from both of us

---

Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/BloomExe/web/controllers/ExternalLinkController.cs, line 85 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-KH5ZUX5gyhk7voPaize:-KH5ZUX5gyhk7voPaizf:-773695400) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e049888b50a473aacddc927b642ddf6403442e41/src/BloomExe/web/controllers/ExternalLinkController.cs#L85)):*
"an fragment" -> "any fragment"?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-:-KH5ZAd5owa0IFdmqNUF:1817474738)*
<!-- Sent from Reviewable.io -->

hatton commented:
Review status: 4 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/BloomBrowserUI/bookEdit/toolbox/readers/leveledReader/leveledReaderToolboxPanel.jade, line 64 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-KH5MesIRO_HR7u_ZQz0:-KHWSsTG9HbIrR2q9b4s:-953748832) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e049888b50a473aacddc927b642ddf6403442e41/src/BloomBrowserUI/bookEdit/toolbox/readers/leveledReader/leveledReaderToolboxPanel.jade#L64)):*
<details><summary><i>Previously, StephenMcConnel (Steve McConnel) wrote…</i></summary>
> Is the &test=foo&blah=boo actually needed here?  what purpose do they serve?</details>
Sorry, that's from a secret society you're not supposed to know about. Removed.

---

*[src/BloomExe/web/controllers/ExternalLinkController.cs, line 27 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-KH5UusQEWWLAJk4ymxi:-KHWTPgcbb6K-8z6az4A:-771183422) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e049888b50a473aacddc927b642ddf6403442e41/src/BloomExe/web/controllers/ExternalLinkController.cs#L27)):*
<details><summary><i>Previously, StephenMcConnel (Steve McConnel) wrote…</i></summary>
> It's probably okay as is, but I would prefer HandleRequest as the name of the method.  Handle initially brings up the image of a magic binary number in my mind.</details>
True! Yuck. Fixed.

---

*[src/BloomExe/web/controllers/ExternalLinkController.cs, line 85 \[r1\]](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051#-KH5ZUX5gyhk7voPaize:-KHWTVyoew7MrPw8yWAQ:-499146009) ([raw file](https://github.com/bloombooks/bloomdesktop/blob/e049888b50a473aacddc927b642ddf6403442e41/src/BloomExe/web/controllers/ExternalLinkController.cs#L85)):*
<details><summary><i>Previously, JohnThomson (John Thomson) wrote…</i></summary>
> "an fragment" -> "any fragment"?</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloomdesktop/1051)*
<!-- Sent from Reviewable.io -->
